### PR TITLE
Enable Python3 checker messages missing for some built-in functions

### PR DESCRIPTION
**Pull request by tiagohonorato**
With the introduction of Python3, some built-in functions changed its return to an iterator type. This PR, related to issue #3466, does the modifications needed to enable checks for `dict.keys, dict.items, filter, range, zip` to warn the use outside an iterator context. In some cases the check had to be disabled as it might be a false positive.

### CODE REVIEW

**Review by richtja — APPROVED** [review]
LGTM, thanks

**Review by willianrampazzo — APPROVED** [review]

File: avocado/plugins/distro.py [hunk 1]
```diff
@@ -353,7 +353,7 @@ def configure(self, parser):
                                  parser=parser,
                                  long_arg='--distro-def-path')
 
-        type_choices = DISTRO_PKG_INFO_LOADERS.keys()
+        type_choices = tuple(DISTRO_PKG_INFO_LOADERS.keys())
```
Comment (beraldoleal) [inline avocado/plugins/distro.py:356]: Any reason for using `tuple` instead of `list` here?

Comment (tiagohonorato) [reply avocado/plugins/distro.py:356]: It was just to follow the docs for `settings.register_option()`, but `list` should work fine as well from what I can remember.

**Review by clebergnu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**